extensions/README.md

```diff
@@ -0,0 +1,373 @@
+[CoRIM
+specification](https://www.ietf.org/archive/id/draft-ietf-rats-corim-02.htm)
+may be extended by CoRIM Profiles documented in other specifications at well defined points identified in the
+base CoRIM spec using CDDL extension sockets. This implementation, likewise, allows
+dependent code to register extension types. This is done via three distinct
+extension mechanisms:
+
+1. Structures that allow extensions embed an `Extensions` object, which allows
+   registering a user-provided struct. The fields of that struct will be
+   treated as extension fields to embedding structure. This corresponds to the
+   map-extension sockets in the spec.
+2. Some fields may have values one of a set of pre-determined types (e.g. an
+   instance ID may be either a UUID or UEID). These (mostly) correspond to
+   type-choice sockets in the CoRIM spec. The set of allowed types for a field
+   may be extended by registering a factory function for a new type, using the
+   registration function associated with the type choice.
+3. A couple of type-choice sockets (`$tag-rel-type-choice`,
+   `$corim-role-type-choice` and `$comid-role-type-choice`) define what, in
+   effect, are extensible enums. They allow providing additional values, rather
+   than types. This implementation provides registration functions for new
+   values for those types.
+
+> [!NOTE]
+> Some of the definitions in the CoRIM spec are "imported" from the CoSWID
+> spec.  Some of those definitions also feature extension sockets. However, as
+> they are defined in a different spec (and are not implemented in this
+> library), they cannot be extended using the above mechanisms.
+
```

Comment:
```suggestion
> [!NOTE]
> CoRIM specification also "imports" CDDL from the CoSWID spec. Some of 
> these COSWID CDDL definitions also feature extension sockets. 

> However, as they are defined in a different spec and are implemented 
> in `veraison/coswid` library, they cannot be extended 
> using the extension feature provided in the CoRIM Library. The extension 
> support in CoRIM library is applicable ONLY to CoRIM  and CoMID Maps and 
> Types choices
```

Reply:
Request please review an attempt to clarify the Note point.

Feel free to edit my text to make it more clear, if you feel so!